pkg/detector/detector.go

```diff
@@ -274,18 +274,7 @@ func (d *ResourceDetector) Reconcile(key util.QueueKey) error {
 // the specified apis will be ignored as well.
 //
 // If '--skipped-propagating-namespaces' is specified, all APIs in the skipped-propagating-namespaces will be ignored.
-func (d *ResourceDetector) EventFilter(obj interface{}) bool {
-	key, err := ClusterWideKeyFunc(obj)
-	if err != nil {
-		return false
-	}
-
-	clusterWideKey, ok := key.(keys.ClusterWideKey)
-	if !ok {
-		klog.Errorf("Invalid key")
-		return false
-	}
-
+func (d *ResourceDetector) EventFilter(clusterWideKey keys.ClusterWideKey) bool {
```

Comment:
I guess the `EventFilter` should be renamed now, as it not handling the `event` anymore, but checks if the specified object should be prevented from propagating. So, how about `SkippedFromPropagating`? 

Remember to update the comments and take care of the function body.